crates/matrix-sdk/src/event_cache/room/mod.rs

```diff
@@ -611,30 +614,52 @@ mod private {
     }
 
     impl RoomEventCacheState {
+        async fn try_reload_linked_chunk(
+            room: &RoomId,
+            locked: &EventCacheStoreLockGuard<'_>,
+        ) -> Result<Option<LinkedChunk<DEFAULT_CHUNK_CAPACITY, Event, Gap>>, EventCacheError>
+        {
+            let raw_chunks = locked.reload_linked_chunk(room).await?;
+
+            let mut builder = LinkedChunkBuilder::from_raw_parts(raw_chunks.clone());
+
+            builder.with_update_history();
+
+            Ok(builder.build().map_err(|err| {
+                // Show a debug string representing the known chunks.
+                if tracing::enabled!(tracing::Level::TRACE) {
+                    trace!("couldn't build a linked chunk from the raw parts. Raw chunks:");
+                    for line in raw_chunks_debug_string(raw_chunks) {
+                        trace!("{line}");
+                    }
+                }
+
+                EventCacheStoreError::InvalidData { details: err.to_string() }
```

Comment:
Shouldn't we erase the storage if we can't deserialize it? What happens if we create a new linked chunk and continue to save it in the storage over a previous one?

Reply:
What you're describing is exactly what happens:

- `try_reload_linked_chunk` fails with an `Err` result
- the caller in `RoomEventCacheState::new()` will log the error, clear storage for the room, and restart with an empty linked chunk